src/Loggers/LogGraph.jl

```diff
@@ -74,10 +45,40 @@ function graph_summary(g, nodelabel, nodeop, nodedevice, nodevalue)
 		elseif x == nothing
 			#donothing
 		else
-			@error "unhandled nodevalue type $(typeof(x))"
+			throw("unhandled nodevalue type $(typeof(x))")
 		end
 		node = NodeDef(name = name, op = op, input = input, device = device, attr = attr)
 		push!(nodes, node)
 	end
 	GraphDef(node = nodes)
 end
+
+function jltype2tf(dtype::DataType)
+	nodetype =
+	dtype == nothing			? _DataType.DT_INVALID :
+	dtype == UInt8				? _DataType.DT_UINT8 :
+	dtype == UInt16				? _DataType.DT_UINT16 :
+	dtype == UInt32				? _DataType.DT_UINT32 :
+	dtype == UInt64				? _DataType.DT_UINT64 :
+	dtype == Int8				? _DataType.DT_INT8 :
+	dtype == Int16				? _DataType.DT_INT16 :
+	dtype == Int32				? _DataType.DT_INT32 :
+	dtype == Int64				? _DataType.DT_INT64 :
+	dtype == Float16			? _DataType.DT_BFLOAT16 :
+	dtype == Float32			? _DataType.DT_FLOAT :
+	dtype == Float64			? _DataType.DT_DOUBLE :
+	dtype <: AbstractString		? _DataType.DT_STRING :
+	dtype == Bool				? _DataType.DT_BOOL :
+	dtype ∈ [Complex{Float32},
+			Complex{Float16},
+			Complex{UInt8},
+			Complex{UInt16},
+			Complex{UInt32},
+			Complex{Int8},
+			Complex{Int16},
+			Complex{Int32}]		? _DataType.DT_COMPLEX64 :
+	dtype ∈ [Complex{Float64},
+			Complex{UInt64},
+			Complex{Int64}]		? _DataType.DT_COMPLEX128 :
+	throw("Unknown Datatype: $dtype")
```

Comment:
I think you meant to throw an exception type,
not a string.

Reply:
I don't think throwing anything is necessary here since `_DataType.DT_INVALID` can be used for unhandled datatypes. Is this good?
```
function jltype2tf(dtype::DataType)
	nodetype =
	dtype == UInt8				? _DataType.DT_UINT8 :
	dtype == UInt16				? _DataType.DT_UINT16 :
	dtype == UInt32				? _DataType.DT_UINT32 :
	dtype == UInt64				? _DataType.DT_UINT64 :
	dtype == Int8				? _DataType.DT_INT8 :
	dtype == Int16				? _DataType.DT_INT16 :
	dtype == Int32				? _DataType.DT_INT32 :
	dtype == Int64				? _DataType.DT_INT64 :
	dtype == Float16			? _DataType.DT_BFLOAT16 :
	dtype == Float32			? _DataType.DT_FLOAT :
	dtype == Float64			? _DataType.DT_DOUBLE :
	dtype <: AbstractString		? _DataType.DT_STRING :
	dtype == Bool				? _DataType.DT_BOOL :
	dtype ∈ [Complex{Float32},
			Complex{Float16},
			Complex{UInt8},
			Complex{UInt16},
			Complex{UInt32},
			Complex{Int8},
			Complex{Int16},
			Complex{Int32}]		? _DataType.DT_COMPLEX64 :
	dtype ∈ [Complex{Float64},
			Complex{UInt64},
			Complex{Int64}]		? _DataType.DT_COMPLEX128 :
	_DataType.DT_INVALID
end
```
or should it `throw(Core.TypeError(Symbol(jltype2tf), "graph_summary", Union{Bool, Number, AbstractString}, dtype))`? Not sure about the syntax of expected argument.

Reply:
I think using `DT_INVALID` is fine.